st2common/bin/st2ctl

```diff
@@ -0,0 +1,197 @@
+}
+
+function validate_in_components() {
+  COM=${1}
+
+  if [ -z ${COM} ]; then
+    echo -e "\e[31mError: Component is required! \e[0m\n"
+    print_usage
+    exit 1
+  fi
+
+  if [[ " ${COMPONENTS} " != *" ${COM} "* ]]; then
+    echo -e "\e[31mError: Invalid component provided: ${COM} \e[0m\n"
+    print_usage
+    exit 1
+  fi
+}
+
+function st2start() {
+  for COM in ${COMPONENTS}; do
+    service ${COM} start
+  done
+}
+
+function st2stop() {
+  for COM in ${COMPONENTS}; do
+    service ${COM} stop
+  done
+}
+
+# Next candidate for removal
+# after we implement native `service st2component reopen` (nginx example) for each component
+function reopen_component_log_files() {
+  COM=${1}
+
+  PID=`ps ax | grep -v grep | grep -v st2ctl | grep "${COM}" | awk '{print $1}'`
+  if [[ ! -z ${PID} ]]; then
+    for p in ${PID}; do
+      echo "Sending SIGUSR1 to ${COM} PID: ${p}"
+      kill -USR1 ${p}
+    done
+  else
+    echo "${COM} is not running"
+    return 1
+  fi
+}
+
+function register_content() {
+  ALLOWED_REGISTER_FLAGS='--register-all --register-actions --register-aliases --register-policies --register-rules --register-sensors --verbose'
+  DEFAULT_REGISTER_FLAGS='--register-actions --register-aliases --register-sensors'
+  flags="${@}"
+
+  if [ ! -z ${1} ]; then
+    for flag in ${flags}; do
+      if [[ " ${ALLOWED_REGISTER_FLAGS} " != *" $flag "* ]]; then # argument not allowed
+        echo -e "\e[31mError: Invalid flag provided: ${flag} \e[0m\n"
+        print_usage
+        exit 1
+      fi
+    done
+  fi
+
+  if [ -z ${1} ]; then
+    REGISTER_FLAGS=${DEFAULT_REGISTER_FLAGS}
+  elif [ ${1} == '--verbose' ] && [ -z ${2} ]; then
+    REGISTER_FLAGS="$DEFAULT_REGISTER_FLAGS ${1}"
+  else
+    REGISTER_FLAGS=${flags}
+  fi
+
+  echo "Registering content...[flags = ${REGISTER_FLAGS}]"
+  st2-register-content --config-file ${STANCONF} ${REGISTER_FLAGS}
+}
+
+function clean_db() {
+  echo "Dropping st2 Database..."
+  mongo st2 --eval "db.dropDatabase();"
+}
+
+function clean_logs() {
+  echo "Cleaning st2 Logs..."
+  rm -Rf /var/log/st2/*
+}
+
+function getpids() {
+  echo "##### st2 components status #####"
+
+  for COM in ${COMPONENTS}; do
+    if [[ "${COM}" == "st2web" && -z "${ST2_DISABLE_HTTPSERVER}" ]]; then
+      PID=`ps ax | grep -v grep | egrep "SimpleHTTPServer $WEBUI_PORT\$" | awk '{print $1}'`
+    elif [[ "${COM}" == "mistral" ]]; then
+      PID=`ps ax | grep -v grep | grep -v postgres | grep "${COM}" | awk '{print $1}'`
+    else
+      PID=`ps ax | grep -v grep | grep -v st2ctl | grep "${COM}" | awk '{print $1}'`
+    fi
+
+    if [[ ! -z ${PID} ]]; then
+      for p in ${PID}; do
+        echo "${COM} PID: ${p}"
+      done
+    else
+      echo "${COM} is not running."
+    fi
+  done
+}
+
+
+case ${1} in
+  start)
+    st2start
+    getpids
+    ;;
+  stop)
+    st2stop
+    ;;
+  restart)
+    st2stop
+    sleep 1
+    st2start
+    getpids
+    ;;
+  restart-component)
```

Comment:
Used: in [`packs.restart_component`](https://github.com/StackStorm/st2/blob/2f6465f366bc22c8adff79c031acd0a3b3837728/contrib/packs/actions/restart_component.yaml#L13)